mesh/mesh.hpp

```diff
@@ -1523,8 +1533,8 @@ std::ostream &operator<<(std::ostream &out, const Mesh &mesh);
 class GeometricFactors
 {
 public:
-   const Mesh *mesh;
-   const IntegrationRule *IntRule;
+   const Mesh &mesh;
+   const IntegrationRule &IntRule;
```

Comment:
@samuelpmishLLNL Do you know if changing the `IntegrationRule` pointer to a reference would have been enough to catch your issue?

Reply:
I'm probably the wrong person to ask about this, because I have little to no understanding of the lifetime requirements of `mfem::IntegrationRule`. In essence, my lack of understanding on this subject is the motivation for creating issue #2106.

Reply:
I don't think changing pointers to references can help catch the original issue in #2106.

For mixed meshes, instead of the `IntegrationRule`, we'll need to work with a `QuadratureSpace`, so `IntRule` will be unused and so it will be more natural to have a pointer which can be set to `nullptr`.

Also this change to references breaks backward compatibility in a public class, which I don't like. The same is true for the modified constructor.